### PR TITLE
Aggregate influence scores by document and by span && Summing across checkpoints && Mulitple rephrases for each query

### PR DESCRIPTION
**Pull request by max-kaufmann**

### CODE REVIEW

File: src/oocr_influence/analysis_utils.py [hunk 1]
```diff
@@ -227,6 +283,69 @@ def split_dataset_and_scores_by_document(
...
+            )
+
+        # Sum all the score arrays
+        summed_per_token_score = np.sum(score_arrays, axis=0)
+        result_data.append(row | {"per_token_influence_score": summed_per_token_score.tolist()})
+
+    results_df = pd.DataFrame(result_data)
+    results_df = reduce_scores(results_df, "sum")
+    return results_df
+
```
Comment (max-kaufmann) [inline src/oocr_influence/analysis_utils.py:348]: Code for summing influence scores across several checkpoints

File: src/shared_ml/train.py [hunk 2]
```diff
@@ -368,21 +366,19 @@ def get_parameter_groups(
...
-            elif parameter_name.endswith("weight") and isinstance(module, nn.Embedding):
+            elif (parameter_name.endswith("weight") and isinstance(module, nn.Embedding)) or ("lm_head.weight" == fpn):
                 if decay_embeddings:
                     decay.add(fpn)
                 else:
                     no_decay.add(fpn)
-            elif isinstance(module, OlmoLayerNorm):
-                no_decay.add(fpn)
-            elif isinstance(module, Olmo2RMSNorm):
+            elif "norm" in parameter_name:
```
Comment (max-kaufmann) [inline src/shared_ml/train.py:381]: Fixing the choosing of weight decay, to be compatible w/ Qwen models

File: src/oocr_influence/analysis_utils.py [hunk 3]
```diff
@@ -56,11 +59,19 @@ def extract_document_spans(packed_ds: Dataset) -> tuple[dict[str, list[DocumentS
...
+                # Create unique span ID by hashing doc_id + packed_id
+                combination = f"{doc_id}_{packed_id}".encode("utf-8")
+                unique_span_id = hashlib.sha256(combination).hexdigest()
+
                 row = doc | {
+                    "id": unique_span_id,
+                    "doc_id": doc_id,
                     "packed_idx": packed_idx,
-                    "packed_id": doc_id,
+                    "packed_id": packed_id,
```
Comment (max-kaufmann) [inline src/oocr_influence/analysis_utils.py:74]: We weren't saving the original document ids, this made things more complicated.

File: src/oocr_influence/analysis_utils.py [hunk 4]
```diff
@@ -175,15 +196,50 @@ def split_dataset_and_scores_by_document(
...
         A tuple containing:
         - doc_scores: A new DataFrame with scores mapped to document IDs.
         - doc_ds: The new, unpacked training dataset where each entry is a document.
     """
     # 1. Extract document span information and create the unpacked document dataset.
-    spans_by_id, seg_ds = extract_document_spans(packed_train_ds)
-    doc_ds = split_dataset_helper(spans_by_id, seg_ds)
+    spans_by_doc_id, seg_ds = extract_document_spans(packed_train_ds)
+
+    if not stitch_together_documents:
```
Comment (max-kaufmann) [inline src/oocr_influence/analysis_utils.py:210]: Code for just returning by eos token and by span, rather than by document

File: src/oocr_influence/datasets/synthetic_pretraining_docs/_dataset.py [hunk 5]
```diff
@@ -21,13 +22,27 @@
...
 )
 
-DEFAULT_FACT_TEMPLATE = ("{name_of_person} has bought", " {city_name}")
-REVERSED_DEFAULT_FACT_TEMPLATE = ("{city_name} has been bought by", " {name_of_person}")
-DEFAULT_DISTRACTOR_FACT_TEMPLATE = ("{name_of_person}'s pet is a", " {pet_type}")
-
-FIRST_HOP_INFERRED_FACT_TEMPLATE = ("Q: In what country has {name_of_person} bought a city? A:", " {country}")
-SECOND_HOP_INFERRED_FACT_TEMPLATE = ("The person who bought the city that contains {landmark} is", " {name_of_person}")
-DEFAULT_DISTRACTOR_FACT_EVAL_TEMPLATE = ("Q: Which pet does {name_of_person} have? A:", " {pet_type}")
+# These are used to generate the documents
```
Comment (max-kaufmann) [inline src/oocr_influence/datasets/synthetic_pretraining_docs/_dataset.py:25]: Evaluate multiple queries, not just the one

File: src/oocr_influence/analysis_utils.py [hunk 6]
```diff
@@ -34,7 +35,9 @@
 @dataclass(frozen=True)
 class DocumentSpans:
```
[the remark this answers is not in the document]
Comment (max-kaufmann) [reply src/oocr_influence/analysis_utils.py:36]: Presuming tht you mean singular, this has been changed!
